yt/fields/field_info_container.py

```diff
@@ -470,6 +471,8 @@ def check_derived_fields(self, fields_to_check=None):
             fi = self[field]
             try:
                 fd = fi.get_dependencies(ds=self.ds)
+            except UnitConversionError:
```

Comment:
I have a suspicion that this is going to cause a lot of heartache.  I do think it's the right thing to do.

One way we might make it an awful lot easier would be to make it much more obvious how to have the units deduced.  As of right now, I don't think that's possible.

Reply:
> One way we might make it an awful lot easier would be to make it much more obvious how to have the units deduced. As of right now, I don't think that's possible.

at what level ? within the code base or from the user perspective ?